ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+
+            if (optUcteNodeCode1.isPresent() && optUcteNodeCode2.isPresent() && optUcteNodeCode3.isPresent() && optUcteNodeCode4.isPresent()) {
+                ArrayList<UcteNodeCode> ucteNodeCodeList = new ArrayList<>();
+                ucteNodeCodeList.add(optUcteNodeCode1.get());
+                ucteNodeCodeList.add(optUcteNodeCode2.get());
+                ucteNodeCodeList.add(optUcteNodeCode3.get());
+                ucteNodeCodeList.add(optUcteNodeCode4.get());
+
+                ucteNodeCodeList.forEach(ucteNodeCode ->
+                        createXnodeFromTieLine(ucteNetwork, ucteNodeCode, line)
+                );
+
+                String elementName1 = line.getProperties().getProperty(UcteImporter.ELEMENT_NAME_PROPERTY_KEY + "_1", null);
+                String elementName2 = line.getProperties().getProperty(UcteImporter.ELEMENT_NAME_PROPERTY_KEY + "_2", null);
+
+                UcteElementId ucteElementId1 = new UcteElementId(ucteNodeCodeList.get(0), ucteNodeCodeList.get(1), line.getId().charAt(18));
+                UcteElementId ucteElementId2 = new UcteElementId(ucteNodeCodeList.get(2), ucteNodeCodeList.get(3), line.getId().charAt(40));
+
+                UcteLine ucteLine1 = new UcteLine(ucteElementId1, UcteElementStatus.REAL_ELEMENT_IN_OPERATION, (float) line.getR() / 2,
+                        (float) line.getX() / 2, (float) line.getB1(), (int) line.getCurrentLimits1().getPermanentLimit(), elementName1);
+                UcteLine ucteLine2 = new UcteLine(ucteElementId2, UcteElementStatus.REAL_ELEMENT_IN_OPERATION, (float) line.getR() / 2,
+                        (float) line.getX() / 2, (float) line.getB1(), (int) line.getCurrentLimits1().getPermanentLimit(), elementName2);
+
+                ucteNetwork.addLine(ucteLine1);
+                ucteNetwork.addLine(ucteLine2);
+            } else {
+                createTieLineWithGeneratedIds(ucteNetwork, line);
+            }
+        } else {
+            createTieLineWithGeneratedIds(ucteNetwork, line);
+        }
+    }
+
+    /**
+     * Creates an UCTE compliant id for the {@link UcteLine}, creates the two UcteLine corresponding to the tie line and adds it to the UcteNetwork
+     *
+     * @param ucteNetwork The target UcteNetwork
+     * @param line The Tie line with the non-compliant UCTE id
+     * @see UcteExporter#createUcteNodeCode
+     * @see UcteExporter#generateUcteElementId(String, UcteNodeCode, UcteNodeCode, Character)
+     */
+    void createTieLineWithGeneratedIds(UcteNetwork ucteNetwork, Line line) {
+        String xnodeCode = xnodeCodeForLine(line);
+        VoltageLevel terminal1VoltageLevel = line.getTerminal1().getVoltageLevel();
+        VoltageLevel terminal2VoltageLevel = line.getTerminal2().getVoltageLevel();
+        Optional<Country> substation1Country = terminal1VoltageLevel.getSubstation().getCountry();
+        Optional<Country> substation2Country = terminal2VoltageLevel.getSubstation().getCountry();
```

Comment:
Change the name of the variables to voltageLevel1, voltageLevel2, country1 and country2